include/octomap/math/Vector3.h:268:5: warning: non-void function does not return a value in all control paths [-Wreturn-type]

QA warning with clang-19 :
```
In file included from octomap-1.10.0/octomap/src/math/Vector3.cpp:34:
octomap-1.10.0/octomap/include/octomap/math/Vector3.h:268:5: warning: non-void function does not return a value in all control paths [-Wreturn-type]
  268 |     }
      |     ^
```

--- a/octomap/include/octomap/math/Vector3.h
+++ b/octomap/include/octomap/math/Vector3.h
@@ -265,6 +265,8 @@ namespace octomath {
         else
           return false;
       }
+      // all equal
+      return false;
     }
 
     /// @return length of the vector ("L2 norm")

--- a/octomap/src/testing/unit_tests.cpp
+++ b/octomap/src/testing/unit_tests.cpp
@@ -52,6 +52,15 @@ int main(int argc, char** argv) {
     EXPECT_FLOAT_EQ (rotation.x(), 1.2750367);
     EXPECT_FLOAT_EQ (rotation.y(), (-1.1329513));
     EXPECT_FLOAT_EQ (rotation.z(), 0.30116868);
+    
+    
+    Vector3 ones2 = *ones;
+    
+    // test comparison
+    EXPECT_TRUE(*ones < *twos);
+    EXPECT_FALSE(*twos < *ones);
+    EXPECT_FALSE(*ones < ones2);
+    EXPECT_FALSE(ones2 < *ones);
   
   // ------------------------------------------------------------
   } else if (test_name == "MathPose") {

